src/reporter/github.test.js

```diff
@@ -303,48 +288,38 @@ describe('GitHub', function () {
         await github.createOrUpdateIssue(ISSUE_TO_CREATE);
       });
 
+      afterEach(() => {
+        github.issuesCache.delete(ISSUE_TO_CREATE.title);
+      });
+
       it('creates the issue', () => {
         expect(createIssueScope.isDone()).to.be.true;
       });
     });
 
     context('when the issue already exists', () => {
-      const ISSUE = {
-        title: 'Existing Issue',
-        description: 'New comment',
-        label: 'location',
-      };
-
       context('when issue is closed', () => {
         let updateIssueScope;
         let addCommentScope;
 
-        const GITHUB_RESPONSE_FOR_EXISTING_ISSUE = {
-          number: 123,
-          title: ISSUE.title,
-          description: ISSUE.description,
-          labels: [{ name: 'selectors' }],
-          state: GitHub.ISSUE_STATE_CLOSED,
-        };
-
         before(async () => {
-          nock('https://api.github.com')
-            .get('/repos/owner/repo/issues')
-            .query(true)
-            .reply(200, [GITHUB_RESPONSE_FOR_EXISTING_ISSUE]);
-
           updateIssueScope = nock('https://api.github.com')
-            .patch(`/repos/owner/repo/issues/${GITHUB_RESPONSE_FOR_EXISTING_ISSUE.number}`, { state: GitHub.ISSUE_STATE_OPEN, labels: ['location'] })
+            .patch(`/repos/owner/repo/issues/${EXISTING_CLOSED_ISSUE.number}`, { state: GitHub.ISSUE_STATE_OPEN, labels: ['location'] })
             .reply(200);
 
           addCommentScope = nock('https://api.github.com')
-            .post(`/repos/owner/repo/issues/${GITHUB_RESPONSE_FOR_EXISTING_ISSUE.number}/comments`, { body: ISSUE.description })
+            .post(`/repos/owner/repo/issues/${EXISTING_CLOSED_ISSUE.number}/comments`, { body: EXISTING_CLOSED_ISSUE.description })
             .reply(200);
 
-          await github.createOrUpdateIssue(ISSUE);
+          await github.createOrUpdateIssue({ title: EXISTING_CLOSED_ISSUE.title, description: EXISTING_CLOSED_ISSUE.description, label: 'location' });
         });
 
-        it('reopens the issue and its labels', () => {
+        after(() => {
+          github.issuesCache.delete(EXISTING_CLOSED_ISSUE);
+          github.issuesCache.set(EXISTING_CLOSED_ISSUE.title, EXISTING_CLOSED_ISSUE);
```

Comment:
I don't understand these: we delete an entry from the cache and then set it back in? 🤔

Reply:
Yes because this entry in the cache was modified by the test